client/components/data/query-email-owner-username/index.ts

```diff
@@ -0,0 +1,52 @@
+import { isTitanMail, isGoogleWorkspace } from '@automattic/calypso-products';
+import { useSelector } from 'react-redux';
+import { useQuerySitePurchases } from 'calypso/components/data/query-site-purchases';
+import useUsersQuery from 'calypso/data/users/use-users-query';
+import { getSitePurchases } from 'calypso/state/purchases/selectors';
+import type { SiteDetails } from '@automattic/data-stores';
+import type { InfiniteData } from 'react-query';
+
+type User = {
+	ID: number;
+	linked_user_ID: number;
+	login: string;
+};
+
+type UsersData = {
+	users: User[];
+};
+
+export function useEmailOwnerUserName(
+	selectedSite: SiteDetails | null | undefined,
+	domainName: string
+): string {
+	useQuerySitePurchases( selectedSite?.ID ?? -1 );
+
+	const purchases = useSelector( ( state ) => getSitePurchases( state, selectedSite?.ID ) );
+
+	const emailSubscription = purchases.find(
+		( purchase ) =>
+			( isTitanMail( purchase ) || isGoogleWorkspace( purchase ) ) &&
+			purchase.meta === domainName &&
+			purchase.active
+	);
+
+	const { data, isLoading } = useUsersQuery(
+		selectedSite?.ID,
+		{},
+		{
+			enabled: emailSubscription !== undefined,
+		}
+	);
+
+	if ( isLoading || ! emailSubscription ) {
+		return '';
+	}
+
+	const teams = data as InfiniteData< UsersData > & UsersData;
+	const ownerUser = teams.users?.find(
```

Comment:
I think we should have an equivalent comment here, like we added in the other file.